lib/solveEquation/stepThrough.js

```diff
@@ -183,11 +279,25 @@ function addSimplificationSteps(steps, equation, debug=false) {
   let oldEquation = equation.clone();
 
   const leftSteps = simplifyExpressionNode(equation.leftNode, false);
+  const simplifiedLeftNode = leftSteps.length !== 0
```

Comment:
And if it's easy to remove some of the redundancy via a helper function, do so, otherwise not a huge deal.